packages/nimble-components/src/rich-text/specs/mention-hld.md

```diff
@@ -0,0 +1,617 @@
+# Rich Text Component Mention Support HLD
+
+## Problem Statement
+
+There is a requirement to `@mention` a user in `nimble-rich-text-editor` and notify them through an email in comments feature. Once the user is
+mentioned in the editor, it should be loaded back to the viewer with the special formatting text in the `nimble-rich-text-viewer`.
+
+### Out of Scope of this Spec
+
+1.  Notifying the mentioned users through an email in comments feature is part of the backend services
+2.  Adding a "Load More" or "Search Server" button to dynamically load more data in the `@mention` dropdown
+in editor.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Rich Text Editor and Viewer README](./README.md)
+-   [Comments `@mention` mockup](https://www.figma.com/file/Q5SU1OwrnD08keon3zObRX/SystemLink-orig?type=design&node-id=7248-114254&mode=design&t=y3JtM3aT77Aw0xjK-0)
+-   [Mention users in comments - Requirement doc](https://dev.azure.com/ni/DevCentral/_git/Skyline?path=/docs/design-documents/Platform/Requirements/Mention-users-in-comments.md&_a=preview)
+-   [Work Item in AzDo](https://dev.azure.com/ni/DevCentral/_workitems/edit/2464517): Support for @mention in comments
+
+## Design
+
+### `nimble-rich-text-editor`
+
+The rich text editor includes this feature for tagging or mentioning a user by inserting the **"@"** symbol within the editor.
+This action triggers a dropdown list, allowing users to select a user. To provide the rich text editor component with the
+necessary user information for populating this dropdown, users can pass the user details through the configuration elements outlined below:
```

Comment:
nit: "users" has multiple meanings so we often prefer "client" to describe the developer configuring the component.
```suggestion
necessary user information for populating this dropdown, clients can pass the user details through the configuration elements outlined below:
```